### PR TITLE
adds investments

### PR DESCRIPTION
**Pull request by rayraycano**
* /investments/holdings/get
* /investments/transactions/get

### CODE REVIEW

**Comment by rayraycano** [comment]
@pbernasconi

File: plaid/holdings.go [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+	InstitutionID          string  `json:"institution_id"`
+	ProxySecurityID        string  `json:"proxy_security_id"`
+	Name                   string  `json:"name"`
+	TickerSymbol           string  `json:"ticker_symbol"`
+	IsCashEquivalent       bool    `json:"is_cash_equivalent"`
+	Type                   string  `json:"type"`
+	ClosePrice             float64 `json:"close_price"`
+	ClosePriceAsOf         string  `json:"close_price_as_of"`
+	ISOCurrencyCode        string  `json:"iso_currency_code"`
+	UnofficialCurrencyCode string  `json:"unofficial_currency_code"`
```
Comment (pbernasconi) [inline plaid/holdings.go]: Why do we have `UnofficialCurrencyCode` at all? Developers might find that confusing

Comment (rayraycano) [reply plaid/holdings.go]: We've documented this in our docs that we've sent out to customers. This is primarily intended for securities like cryto (BTC) or rewards points at certain institutions. Is this something you think we should remove?

File: plaid/investment_transactions.go [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+	APIResponse
+	Item                        Item                    `json:"item"`
+	Accounts                    []Account               `json:"accounts"`
+	InvestmentTransactions      []InvestmentTransaction `json:"investment_transactions"`
+	Securities                  []Security              `json:"securities"`
+	TotalInvestmentTransactions int                     `json:"total_investment_transactions"`
+}
+
+type GetInvestmentTransactionsOptions struct {
+	StartDate  string
```
Comment (pbernasconi) [inline plaid/investment_transactions.go]: this + `EndDate` shouldn't be in the options struct

Comment (pbernasconi) [reply plaid/investment_transactions.go]: oh wait - nvm this is the param object passed in, ignore what I said

File: plaid/investment_transactions.go [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+type getInvestmentTransactionsRequestOptions struct {
+	AccountIDs []string `json:"account_ids"`
+	Count      int      `json:"count"`
+	Offset     int      `json:"offset"`
+}
+
+// GetInvestmentTransactionsWithOptions retrieves user-authorized investment transaction data for investment-type accounts.
+// See https://plaid.com/docs/api/#investment-transactions.
+func (c *Client) GetInvestmentTransactionsWithOptions(accessToken string, options GetInvestmentTransactionsOptions) (resp GetInvestmentTransactionsResponse, err error) {
+	if options.StartDate == "" || options.EndDate == "" {
```
Comment (pbernasconi) [inline plaid/investment_transactions.go]: For good measure can we check that `accessToken  != ""` - similar to `GetHoldings`?

File: plaid/investment_transactions.go (hunk 2 above)
Comment (rayraycano) [reply plaid/investment_transactions.go]: It looks like `transactions` follows the same format: https://github.com/plaid/plaid-go/blob/master/plaid/transactions.go#L75
Alternatively we could have them pass it in as arguments to `GetInvestmentsTransactions` -- thoughts?

File: plaid/investment_transactions.go (hunk 3 above)
Comment (rayraycano) [reply plaid/investment_transactions.go]: ⚡️

File: plaid/investment_transactions.go (hunk 2 above)
Comment (rayraycano) [reply plaid/investment_transactions.go]: If we're going to error when options.StartDate/options.EndDate are `""`, I'm fine with making them explicit arguments for the function. I'm also fine with being consistent with `GetTransactionsWithOptions`

Comment (pbernasconi) [reply plaid/investment_transactions.go]: I think we should mimic the other methods in this lib – so matching `GetTransactionsWithOptions` is the best direction IMO

Comment (rayraycano) [reply plaid/investment_transactions.go]: Okay. In that case I'll be leaving `StartDate` and `EndDate` in the `GetInvestmentTransactionsOptions` struct

**Pull request merged**